backend/app/usecase/shortlink/updater_test.go

```diff
@@ -0,0 +1,225 @@
+			shortLinkUpdate: entity.ShortLink{
+				LongLink: validNewLongLink,
+			},
+			relationUsers: []entity.User{
+				{ID: "1"},
+			},
+			relationShortLinks: []entity.ShortLink{
+				{
+					Alias:     "boGp9w35",
+					LongLink:  "https://httpbin.org",
+					UpdatedAt: &now,
+				},
+			},
+			expHasErr: false,
+			expectedShortLink: entity.ShortLink{
+				Alias:    "boGp9w35",
+				LongLink: validNewLongLink,
+			},
+		},
+		{
+			name:  "alias doesn't exist",
+			alias: validNewAlias,
+			shortlinks: shortLinks{
+				"boGp9w35zzzz": entity.ShortLink{
+					Alias:     "boGp9w35zzzz",
+					LongLink:  "https://httpbin.org",
+					UpdatedAt: &now,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "gopher@golang.org",
+			},
+			shortLinkUpdate: entity.ShortLink{
+				LongLink: validNewLongLink,
+			},
+			relationUsers: []entity.User{
+				{ID: "1"},
+			},
+			relationShortLinks: []entity.ShortLink{
+				{
+					Alias:     "boGp9w35",
+					LongLink:  "https://httpbin.org",
+					UpdatedAt: &now,
+				},
+			},
+			expHasErr:         true,
+			expectedShortLink: entity.ShortLink{},
+		},
+		{
+			name:  "long link is invalid",
+			alias: alias,
+			shortlinks: shortLinks{
+				"boGp9w35": entity.ShortLink{
+					Alias:     "boGp9w35",
+					LongLink:  "https://httpbin.org",
+					UpdatedAt: &now,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "gopher@golang.org",
+			},
+			shortLinkUpdate: entity.ShortLink{
+				LongLink: "aaaaaaaaaaaaaaaaaaa",
+			},
+			relationUsers: []entity.User{
+				{ID: "1"},
+			},
+			relationShortLinks: []entity.ShortLink{
+				{
+					Alias:     "boGp9w35",
+					LongLink:  "https://httpbin.org",
+					UpdatedAt: &now,
+				},
+			},
+			expHasErr:         true,
+			expectedShortLink: entity.ShortLink{},
+		},
+		{
+			name:  "short link is not owned by the user",
+			alias: alias,
+			shortlinks: shortLinks{
+				"boGp9w35": entity.ShortLink{
+					Alias:     "boGp9w35",
+					LongLink:  "https://httpbin.org",
+					UpdatedAt: &now,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "gopher@golang.org",
+			},
+			shortLinkUpdate: entity.ShortLink{
+				LongLink: "https://google.com/",
+			},
+			relationUsers: []entity.User{
+				{ID: "2"},
+			},
+			relationShortLinks: []entity.ShortLink{
+				{
+					Alias:     "boGp9w35",
+					LongLink:  "https://httpbin.org",
+					UpdatedAt: &now,
+				},
+			},
+			expHasErr:         true,
+			expectedShortLink: entity.ShortLink{},
+		},
+		{
+			name:  "malicious shortLink update",
+			alias: alias,
+			user: entity.User{
+				Email: "gopher@golang.org",
+			},
+			shortLinkUpdate: entity.ShortLink{
+				LongLink: "http://malware.wicar.org/data/ms14_064_ole_not_xp.html",
+			},
+			expHasErr:         true,
+			expectedShortLink: entity.ShortLink{},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+			blockedHash := map[string]bool{
```

Comment:
```suggestion
			blockedHash := map[string]struct{}{
```
Use empty struct to save some spaces.

Reply:
It was originally changed to bool in a previous code review to avoid the weird `map[string]struct{}{}` syntax. Changing the map value type will force a lot of tests to update. I don't think the byte difference is worth changing it at this point. Maybe in the future?

Reply:
@oatovar Sounds good to me!